var/spack/repos/builtin/packages/kokkos/package.py

```diff
@@ -280,6 +292,11 @@ def cmake_args(self):
                         raise SpackError("Unsupported target: {0}".format(
                             amdgpu_target))
 
+        for arch in self.intel_gpu_arches:
+            keyval = "intel_gpu_arch=%s" % arch
+            if keyval in spec:
+                spack_microarches.append(arch)
```

Comment:
If it is a single valued variant, we can avoid looping and just extract the value.